clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -603,6 +677,12 @@ public synchronized void handleCompletedBatch(ProducerBatch batch, ProduceRespon
 
```

Comment:
nit: If we `maybeUpdateLastAckedSequence` return the last acked sequence, then we could avoid the awkward line above `lastAckedSequence(batch.topicPartition).orElse(-1)`